include/picongpu/fields/laserProfiles/PulseFromSpectrum.hpp

```diff
@@ -0,0 +1,335 @@
+
+                            E_a += a * math::cos(2._X * float_X(PI) * v_k * runTime);
+                            E_b += b * math::sin(2._X * float_X(PI) * v_k * runTime);
+                        }
+
+                        // E(t)-Field derived from Spectrum
+                        float_64 E_t = (E_a + E_b) / float_X(2 * N + 1);
+                        E_t *= Unitless::AMPLITUDE;
+
+                        return E_t;
+                    }
+
+                    /** Device-Side Constructor
+                     *
+                     * @param superCellToLocalOriginCellOffset local offset in cells to current supercell
+                     * @param offsetToTotalDomain offset to origin of global (@todo: total) coordinate system (possibly
+                     * after transform to centered origin)
+                     */
+                    HDINLINE PulseFromSpectrum(
+                        typename FieldE::DataBoxType const& dataBoxE,
+                        DataSpace<simDim> const& superCellToLocalOriginCellOffset,
+                        DataSpace<simDim> const& offsetToTotalDomain,
+                        float3_X const& elong,
+                        float_X const phase,
+                        uint32_t const currentStep)
+                        : BaseFunctor(dataBoxE, superCellToLocalOriginCellOffset, offsetToTotalDomain, elong)
+                        , m_phase(phase)
+                        , m_currentStep(currentStep)
+                    {
+                    }
+
+                    /** device side manipulation for init plane (transversal)
+                     *
+                     * @tparam T_Args type of the arguments passed to the user manipulator functor
+                     *
+                     * @param cellIndexInSuperCell ND cell index in current supercell
+                     */
+                    template<typename T_Acc>
+                    HDINLINE void operator()(T_Acc const&, DataSpace<simDim> const& cellIndexInSuperCell)
+                    {
+                        // coordinate system to global simulation as origin
+                        DataSpace<simDim> const localCell(
+                            cellIndexInSuperCell + this->m_superCellToLocalOriginCellOffset);
+
+                        // transform coordinate system to center of x-z plane of initialization
+                        constexpr uint8_t planeNormalDir = 1u;
+                        DataSpace<simDim> offsetToCenterOfPlane(this->m_offsetToTotalDomain);
+                        offsetToCenterOfPlane[planeNormalDir] = 0; // do not shift origin of plane normal
+                        floatD_X const pos
+                            = precisionCast<float_X>(localCell + offsetToCenterOfPlane) * cellSize.shrink<simDim>();
+                        // @todo add half-cells via traits::FieldPosition< Solver::NumicalCellType, FieldE >()
```

Comment:
I see this is copied from the existing gaussian profile. We probably can easily account for it in all relevant profiles in a separate PR